model/storage/RdsStorage.php

```diff
@@ -236,9 +237,55 @@ protected function saveData(Revision $revision, array $data)
      * @return array
      */
     public function getResourcesUriByQuery(string $query, array $options = [], string $predicate = OntologyRdfs::RDFS_LABEL)
+    {
+        $result = $this->getSelectedResourcesDataByQuery(
+            [self::DATA_RESOURCE],
+            $query,
+            $options,
+            $predicate,
+        );
+
+        $resourcesUri = [];
+        while ($statement = $result->fetch()) {
+            $resourcesUri[] = $statement[self::DATA_RESOURCE];
+        }
+        return $resourcesUri;
+    }
+
+    public function getResourcesDataByQuery(string $query, array $options = [], string $predicate = OntologyRdfs::RDFS_LABEL): array
```

Comment:
Any reason to be public for this method? 

```suggestion
    private function getResourcesDataByQuery(string $query, array $options = [], string $predicate = OntologyRdfs::RDFS_LABEL): array
```